tests/test_utils.py

```diff
@@ -377,3 +380,16 @@ class Meta:
     }
 
     assert included_serializers == expected_included_serializers
+
+def test_get_resource_type_from_serializer_error_message():
+
+    class SerializerWithoutResourceName(serializers.Serializer):
+        something = Field()
+    
+    serializer = SerializerWithoutResourceName()
+        
+    try:
```

Comment:
to simplify use `pytest.raises` instead. See an example [here](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/tests/test_parsers.py#L108).